daliuge-engine/dlg/manager/proc_daemon.py

```diff
@@ -207,6 +211,19 @@ def startDIM(self, nodes):
         logger.info(
             "Started Data Island Drop Manager with PID %d" % (self._dim_proc.pid)
         )
+
+        # Registering the new DIM via zeroconf so it gets discovered
+        # by the Master Manager
+        if self._zeroconf:
+            addrs = utils.get_local_ip_addr()
+            logger.info("Registering this DIM with zeroconf: %s" % addrs)
+            self._nm_info = utils.register_service(
```

Comment:
This result show go into a new `self._dim_info` attribute rather than `self._nm_info`. Like `self._nm_info`, the new attribute should then be used to deregister from zeroconf at DIM-stopping time.